File: cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    // getting change as input
    float change;
    do
    {
        change = get_float("Change owed: ");
    }
    while (change < 0.0);
    int cents = round (change * 100); // rounding to count for error

    int coins = 0;
    int tot = 0;
    
    coins = coins + (cents/25);
    tot = (cents%25);
        
    coins = coins + (tot/10);
    tot = (tot%10);

    coins = coins + (tot/5);
    tot = (tot%5);

    coins = coins + (tot/1);
    tot = (tot%1);
    
    printf("%i\n", coins);
}

File: cash.py
from cs50 import get_float

# getting user input and checking if positive
while True:
    change = get_float('Change owed: ') * 100
    if change > 0:
        break

coins = 0  # counter

for denomination in [25, 10, 5, 1]:  # looping through each available denomination
    coins = coins + int(change / denomination)
    change = change % denomination 
    if change == 0:  # checking if done
        break

print(coins)
